src/supervisor/workload-sanitization.ts

```diff
@@ -14,6 +15,7 @@ export function trimWorkloads<
 export function trimWorkload<
   T extends KubernetesObject & Partial<{ status: unknown; spec: unknown }>,
 >(workload: T): T & { metadata: V1ObjectMeta } {
+  logger.debug(workload, 'workload state before trimming');
```

Comment:
After further considering it, we probably don't want to log the full workload. It will be extremely noisy! I think the logging we do in `trimMetadata()` should be enough

Reply:
Or we can log the `Metadata` before trimming?